docs/streaming.md

```diff
@@ -0,0 +1,95 @@
+# Spark Streaming
+
+Spark Streaming is an extension of the core Spark API that enables scalable, high-throughput, fault-tolerant stream processing of live data streams. Data can be ingested from many sources like Kafka, Flume, Kinesis, or TCP sockets, and can be processed using complex algorithms expressed with high-level functions like map, reduce, join and window. 
```

Comment:
It looks like this paragraph is taken from the Apache Spark docs. Let's add a sentence that gives attribution to that source (http://spark.apache.org/docs/latest/streaming-programming-guide.html)  cc @korry8911